src/components/measurements/measurementsD3.js

```diff
@@ -305,7 +305,12 @@ export const drawMeansForColorBy = (ref, svgData, treeStrainColors) => {
     // 2 x subplotPadding for padding around the overall mean display
     const ySpacing = (layout.subplotHeight - 4 * layout.subplotPadding) / (numberOfColorByAttributes - 1);
     let yValue = layout.subplotPadding;
-    Object.entries(colorByGroups).forEach(([attribute, {color, values}]) => {
+    // Order the color groups by the legend value order so that we have a stable order for the means
+    orderBy(
+      Object.entries(colorByGroups),
+      ([attribute]) => legendValues.indexOf(attribute),
```

Comment:
This doesn't work when the `legendValues` are numerics (`legendValues.indexOf(attribute)=-1`) and thus the ordering isn't working for continuous colour scales (date, antigenic advance etc).

The `legendValues` are inherently ordered, so could we not simply replace the `orderBy` call with the following?

```js
    legendValues.filter((v) => String(v) in colorByGroups)
      .forEach((attribute) => {
        const {color, values} = colorByGroups[attribute];
        ...
```

This same incantation is used in one other place in auspice, so might be worth reviewing that as well (I haven't done so): https://github.com/nextstrain/auspice/blob/2eb4bb35f28375914f638eb894b9eb00c254fce3/src/components/measurements/index.js#L176

Reply:
Ah, thank you for catching that! I'll update to your suggestion. That's much easier to read and understand

Reply:
In the other use of `orderBy` that you pointed out, the `fieldOrder` is not guaranteed to include every single display field. We want to order `displayFields` by the `fieldOrder` first then the remaining fields can fall into their existing order.

Reply:
> In the other use of orderBy that you pointed out, the fieldOrder is not guaranteed to include every single display field. We want to order displayFields by the fieldOrder first then the remaining fields can fall into their existing order.

👍 Sounds good. And since `displayFields` is a list of keys, they won't be numeric (which is the underlying potential bug here).